compiler-plugin/src/main/kotlin/com/google/devtools/ksp/processing/impl/ResolverImpl.kt

```diff
@@ -697,3 +718,38 @@ private fun Name.getNonSpecialIdentifier() :String {
         asString().substring(1)
     }
 }
+
+private inline fun MemberScope.findEnclosedDescriptor(
+    kindFilter: DescriptorKindFilter,
+    crossinline filter: (DeclarationDescriptor) -> Boolean
+) : DeclarationDescriptor? {
+    return getContributedDescriptors(
+        kindFilter = kindFilter
+    ).firstOrNull(filter)
+}
+
+private inline fun ClassDescriptor.findEnclosedDescriptor(
+    kindFilter: DescriptorKindFilter,
+    crossinline filter: (DeclarationDescriptor) -> Boolean
+) : DeclarationDescriptor? {
+    return this.unsubstitutedMemberScope.findEnclosedDescriptor(
+        kindFilter = kindFilter,
+        filter = filter
+    ) ?: this.staticScope.findEnclosedDescriptor(
+        kindFilter = kindFilter,
+        filter = filter
+    )
+}
+
+private fun ClassDescriptor.findEnclosedDescriptor(
+    kindFilter: DescriptorKindFilter,
+    psi: PsiElement
+): DeclarationDescriptor? {
+    return this.unsubstitutedMemberScope.findEnclosedDescriptor(
```

Comment:
Since you are using `firstOrNull` it should make no performance impact if we just combine 2 lists together and call the filter.

Reply:
this method is going away (and it was wrong by not delegating to the ClassDescriptor filter version) so i'm assuming this comment applies to the other one as well?
in that case though i have member scopes that i cannot combine.
I can combine the results of getContributedDescriptors but that would have two unnecessary things:
* it would query staticScope regardless even when we find the result in member scope
* it would create another list to combine the two.

hence, i think it is better to keep the `ClassDescriptor.findEnclosedDescriptor` above as is? (this function is going away)